### PR TITLE
turns taming into a component

### PR DESCRIPTION
**Pull request by ghost**
a lot of simple mob code should be components / elements, this is one of those parts.

this component gives mobs the ability to be tamed, you can add a callback for any special post-tame behavior.

### CODE REVIEW

File: code/modules/events/wizard/magicarp.dm [hunk 1]
```diff
@@ -49,6 +47,8 @@
 	projectiletype = pick(allowed_projectile_types)
 	. = ..()
 
+/mob/living/simple_animal/hostile/carp/cayenne/make_tameable()
+	return
```
Comment (Fikou) [inline code/modules/events/wizard/magicarp.dm:51]:
```suggestion
	return

```

File: code/modules/events/wizard/magicarp.dm [hunk 2]
```diff
@@ -38,9 +38,7 @@
 	health = 50
 	gold_core_spawnable = NO_SPAWN
 	random_color = FALSE
-	food_type = list()
-	tame_chance = 0
-	bonus_tame_chance = 0
+
```
Comment (Fikou) [inline code/modules/events/wizard/magicarp.dm:41]:
```suggestion
```

Comment (Qustinnus) [reply code/modules/events/wizard/magicarp.dm:41]: i need to override this differently but havnt yet which is why i havnt removed these.

File: code/modules/events/wizard/magicarp.dm (hunk 1 above)
Comment (Qustinnus) [reply code/modules/events/wizard/magicarp.dm:51]: ?

**Comment by interestingusernam3** [comment]
Epic compilation fail

**Comment by Qustinnus** [comment]
i sleep

**Pull request closed without merging**